analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.ExceptionCandidate.cs

```diff
@@ -572,5 +572,123 @@ End Sub
             validator.ValidateTagOrder("BeforeTry", "InTry");
             validator.ExitStates.Should().HaveCount(0);
         }
+
+#if NET
+
+        [TestMethod]
+        public void ExceptionCandidate_Index_Array()
+        {
+            const string code = @"
+var tag = ""BeforeTry"";
+try
+{
+    tag = ""InTry"";
+    _ = array[^0];
+}
+catch
+{
+    tag = ""InCatch"";
+}
+tag = ""AfterCatch"";";
+
+            var validator = SETestContext.CreateCS(code, ", int[] array").Validator;
+            validator.ValidateContainsOperation(OperationKindEx.MethodReference);
+            validator.ValidateTagOrder("BeforeTry", "InTry", "InCatch", "AfterCatch");
+            validator.TagStates("InCatch").Should().ContainSingle(x => HasExceptionOfType(x, "IndexOutOfRangeException"));
+            validator.ExitStates.Should().HaveCount(1).And.ContainSingle(x => HasNoException(x));
+        }
+
+        [TestMethod]
+        public void ExceptionCandidate_Index_This()
+        {
+            const string code = @"
+using System;
+
+class Sample
+{
+    public int this[Index i] => 0;
+
+    public void Method()
+    {
+        var tag = ""BeforeTry"";
+        try
+        {
+            tag = ""InTry"";
+            _ = this[^0];
+        }
+        catch
+        {
+            tag = ""UnreachableInCatch"";
+        }
+        tag = ""AfterCatch"";
+    }
+}";
+
+            var validator = CreateCSharpValidator(code);
+            // IImplicitIndexerReferenceOperation is not generated
```

Comment:
To help our future selves to understand why the hell the master failed after updating the VM
```suggestion
            // IImplicitIndexerReferenceOperation is not generated in the current version of Roslyn. It will be generated in 4.4.0.
```

Reply:
How did you find out the Roslyn version? Checked the code history to see when it was introduced?

Reply:
I verified on sharplab, that it produces the expected operation after switching to `main` branch. And then
https://github.com/dotnet/roslyn/blob/412dc4cef33b6bfa1ae8213b76dc0fd0d0942c75/eng/Versions.props#L8-L12

Reply:
We have 4.2.0. It could be also introduced in 4.3.0, but that is not important detail.

Reply:
In tests [we actually use 4.0.0.4-final](https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj#L38). I've tried to update to 4.2.0 but it wasn't straight forward since a lot of tests were failing so I've postponed this for when we will have more time to dedicate on this.